lib/reek/context/method_context.rb

```diff
@@ -35,7 +42,24 @@ def default_assignments
       end
 
       def singleton_method?
-        exp.singleton_method? || visibility == :module_function
+        false
+      end
+
+      def instance_method?
+        true
+      end
+
+      def apply_current_visibility(current_visibility)
+        self.visibility = current_visibility
+      end
+
+      def module_function?
+        visibility == :module_function
+      end
+
+      # @return [Boolean] If the visibility is public or not.
```

Comment:
As much as I love yard doc, I think we can skip comments like this. I mean, the method name literally says this already and it uses the predicate notation as well...